go/tasks/pluginmachinery/k8s/plugin.go

```diff
@@ -91,3 +91,68 @@ type Plugin interface {
 	// Properties desired by the plugin
 	GetProperties() PluginProperties
 }
+
+type PluginAbortOverride interface {
```

Comment:
lets add some comments on this explaining when should you implement this interface and why it exists

Reply:
overall this patch looks good to me, just needs a  comments and explanations to help new plugin authors understand the idea